docs/paper/dev/api/roadmap.md

```diff
@@ -0,0 +1,93 @@
+---
+slug: /dev/roadmap
+---
+
+# Roadmap
+
+Paper offers a rich API that has a wide range of features that can help you unlock the full potential of your server. 
+However, in order to make room for new features and improvements, some of the older APIs will be phased out. This page 
+is intended to document any future API changes that are planned or possible deprecations that may be coming up. 
+
+## Future Plans
+
+
+### Interface ItemStacks
+
+When you create ItemStacks, you create an API representation of an ItemStack.
+However, there are also places where you can obtain an ItemStack that is backed by an NMS object instead. 
+This can lead to inconsistencies and unnecessary upkeep since we need to maintain our own implementation of ItemStack and
+support the NMS backed object using ugly methods due to it not being a plain interface.
+
+In the future, ItemStack will be converted to an interface that allows developers to use an underlying 
+ItemStack directly instead of going through the API implementation.
+
+#### Precautions
+- Avoid using the ItemStack constructor, instead use the static constructors
+    - `ItemStack.of(ItemType.DIAMOND_PICKAXE);`
+      - The ``ItemStack`` constructor will be removed at some point.
+- Avoid directly extending the ``ItemStack`` class.
+  - Custom implementations of this class are not supported
+
+
+## Experimental API
+
+### Teleport Flags
+
+Teleport flags offer a way to teleport entities whilst being able to customize behavior.
+This allows you to do things like teleport players using relative flags and being able to retain passengers.
+
+This API is currently finalized and will be marked as stable in a future release.
+
+#### Player Teleportation
+Teleport a player relatively, preventing velocity from being reset in the X, Y, and Z axis.
+```java
+player.teleport(location,
+            TeleportFlag.Relative.X,
+            TeleportFlag.Relative.Y,
+            TeleportFlag.Relative.Z
+        );
+```
+
+#### Vehicle Teleportation
+Teleport an entity with the `RETAIN_PASSENGERS` flag, allowing its passengers to be transferred with the entity.
+```java
+entity.teleport(location, TeleportFlag.EntityState.RETAIN_PASSENGERS);
+```
+
+## Deprecation Policy
+
+:::warning
+
+It is highly recommended that you avoid using any APIs that are marked for deprecation.
```

Comment:
I would change the wording here to be `marked as deprecated`. Marked "for deprecation" sounds like they will be deprecated in the future. I would also stress that anything scheduled for removal (include 2 examples, one with the deprecated annotation, one with the ApiStatus one) **WILL** be removed in the future, while just being marked as deprecated does not mean it won't be removed, but it isn't currently scheduled. Also talk about fair lead teim being given after something is scheduled for removal and actual removal. And actual removal will only happen on major mc version updates.